app/models/cart_item.rb

```diff
@@ -5,4 +5,22 @@ class CartItem < ApplicationRecord
   belongs_to :product
   validates :price, presence: true
   belongs_to :discount, optional: true
-end
+  validate :check_age_restriction
+
+
+  def check_age_restriction
+    if product.age_restricted? && user&.profile&.age.to_i  < 18
```

Comment:
user&.profile ? every user instance will have profile for sure. right?

Reply:
yes every user will have a profile.I made a associations between user and profile.